### ISSUE TITLE
Supporting GPG for verified commits

### ISSUE THREAD

**Issue by moabu** [google-github-actions/release-please-action#399]
### TL;DR

In protected branches that require all commits to be signed the release actions commit causes the checks to always fail because it doesn't allow to sign the commit `git commit -S -s -m "chore(build): sign the commit"`

### Detailed design

It would be great to either allow to pass in the gpg key and passphrase for the bot being used or to allow another step if possible to configure git and signed the commit although I believe the latter is not possible.

```yaml
    - uses: google-github-actions/release-please-action@v2
      id: release-please
      with:
         gpg_private_key: ${{ secrets.MY_GPG_PRIVATE_KEY }}
         passphrase: ${{ secrets.MY_GPG_PRIVATE_KEY_PASSPHRASE }}
         git_user_signingkey: true
         git_commit_gpgsign: true
         path: ${{ matrix.docker-images }}
         token: ${{ secrets.MYTOKEN }}
         release-type: python
         package-name: ${{ matrix.python-projects }}
         monorepo-tags: true
```

### Additional information
![Screen Shot 2021-11-29 at 3 44 02 AM](https://user-images.githubusercontent.com/47318409/143835982-ae354659-55d2-443f-8e24-0985076ea9fd.png)

**Comment by nickaguilarh** [comment]
Hi there,
Any plan to include this on any milestone?

I'm facing the same issue as @moabu. Using this GH Action to import the GPG Key
[crazy-max/ghaction-import-gpg@v5](https://github.com/crazy-max/ghaction-import-gpg)

**Comment by chingor13** [comment]
The underlying libraries (https://github.com/googleapis/release-please/issues/1314, https://github.com/googleapis/code-suggester/issues/337) needs to support this option which is non-trivial
